### PR TITLE
Remove ParagraphStyle.lineCount, which has been superseded by maxLines

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/7723

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth

**Comment by abarth** [comment]
LGTM

File: lib/ui/text.dart [hunk 1]
```diff
@@ -494,18 +483,17 @@ class ParagraphStyle {
...
              'textAlign: ${   _encoded[0] & 0x02 == 0x02 ? TextAlign.values[_encoded[1]]    : "unspecified"}, '
              'fontWeight: ${  _encoded[0] & 0x04 == 0x04 ? FontWeight.values[_encoded[2]]   : "unspecified"}, '
              'fontStyle: ${   _encoded[0] & 0x08 == 0x08 ? FontStyle.values[_encoded[3]]    : "unspecified"}, '
-             'lineCount: ${   _encoded[0] & 0x10 == 0x10 ? _encoded[4]                      : "unspecified"}, '
+             'maxLines: ${    _encoded[0] & 0x10 == 0x10 ? _encoded[4]                      : "unspecified"}, '
              'fontFamily: ${  _encoded[0] & 0x20 == 0x20 ? _fontFamily                      : "unspecified"}, '
              'fontSize: ${    _encoded[0] & 0x40 == 0x40 ? _fontSize                        : "unspecified"}, '
              'lineHeight: ${  _encoded[0] & 0x80 == 0x80 ? "${_lineHeight}x"                : "unspecified"}, '
-             'maxLines: ${    _encoded[0] & 0x100 == 0x100 ? _maxLines                      : "unspecified"}, '
              'ellipsis: ${    _encoded[0] & 0x200 == 0x200 ? "\"$_ellipsis\""               : "unspecified"}'
```
Comment (Hixie) [inline lib/ui/text.dart]: 200 should be 100 here

Comment (Hixie) [inline lib/ui/text.dart]: there should be a test to catch that...

File: lib/ui/text.dart [hunk 2]
```diff
@@ -445,34 +440,29 @@ class ParagraphStyle {
...
                                         lineHeight,
-                                        maxLines,
                                         ellipsis),
        _fontFamily = fontFamily,
        _fontSize = fontSize,
        _lineHeight = lineHeight,
-       _maxLines = maxLines,
        _ellipsis = ellipsis {
-    assert(lineCount == null);
   }
```
Comment (Hixie) [inline lib/ui/text.dart]: travis' complaint is that you should remove the { } and replace it with a ;

File: lib/ui/text.dart (hunk 1 above)
Comment (jason-simmons) [inline lib/ui/text.dart]: done

**Pull request merged**